conjure-java-jaxrs-client/src/main/java/com/palantir/conjure/java/client/jaxrs/DialogueFeignClient.java

```diff
@@ -0,0 +1,346 @@
+        static final feign.codec.ErrorDecoder INSTANCE = new RemoteExceptionDecoder();
+
+        private static final ObjectMapper MAPPER = ObjectMappers.newClientObjectMapper();
+
+        private static final feign.codec.ErrorDecoder defaultDecoder = new feign.codec.ErrorDecoder.Default();
+
+        private boolean isError(feign.Response response) {
+            return 300 <= response.status() && response.status() <= 599;
+        }
+
+        private Exception parse(feign.Response response) {
+            String body;
+            try {
+                body = toString(response.body().asInputStream());
+            } catch (NullPointerException | IOException e) {
+                UnknownRemoteException exception = new UnknownRemoteException(response.status(), "<unparseable>");
+                exception.initCause(e);
+                return exception;
+            }
+
+            Optional<String> contentType = Optional.ofNullable(
+                            response.headers().get(HttpHeaders.CONTENT_TYPE))
+                    .map(values -> Iterables.getFirst(values, null));
+            if (contentType.isPresent()
+                    && contentType.get().toLowerCase(Locale.ENGLISH).startsWith("application/json")) {
+                try {
+                    SerializableError serializableError = MAPPER.readValue(body, SerializableError.class);
+                    return new RemoteException(serializableError, response.status());
+                } catch (Exception e) {
+                    return new UnknownRemoteException(response.status(), body);
+                }
+            }
+
+            return new UnknownRemoteException(response.status(), body);
+        }
+
+        private static String toString(InputStream body) throws IOException {
+            try (Reader reader = new InputStreamReader(body, StandardCharsets.UTF_8)) {
+                return CharStreams.toString(reader);
+            }
+        }
+
+        @Override
+        public Exception decode(String methodKey, feign.Response response) {
+            if (isError(response)) {
+                return parse(response);
+            }
+            return defaultDecoder.decode(methodKey, response);
+        }
+    }
+
+    private final class FeignRequestEndpoint implements Endpoint {
+        private final feign.Request request;
+        private final String endpoint;
+
+        FeignRequestEndpoint(feign.Request request) {
+            this.request = request;
+            endpoint = getFirstHeader(request, PATH_TEMPLATE).orElse("feign");
+        }
+
+        @Override
+        public void renderPath(Map<String, String> _params, UrlBuilder url) {
+            String target = request.url();
+            Preconditions.checkState(
+                    target.startsWith(baseUrl),
+                    "Request URL must start with base url",
+                    UnsafeArg.of("baseUrl", baseUrl));
```

Comment:
might need to include the `target` here as we presumably always know what the desired baseUrl is?